### DISCUSSION TITLE
Unable to download schema from GitHub

### DISCUSSION THREAD

**Discussion by ben-walker** [dotansimha/graphql-code-generator#5142] [Q&A] [answered]
Hello! I'm trying to point GraphQl Code Generator at a remote schema hosted on GitHub, following the config format outlined here: [https://graphql-code-generator.com/docs/getting-started/schema-field#github](https://graphql-code-generator.com/docs/getting-started/schema-field#github)

But following that format doesn't seem to work - I always get `Error: Unable to download schema from github: [object Object]`.

The GitHub repo is public, so I don't think it's an access issue, although it is "owned" by an organization, maybe that's causing the issue? I also see mention of a `token` in `@graphql-tools/github-loader`, should I be authenticating with GitHub somehow in my `codegen.yaml` file?

Thanks for all the great work!

**Comment by dotansimha** [comment, marked as answer]
This might be an issue related to `graphql-tools` and this specific loader. 
@ben-walker can you please share a reproduction of the issue? 

fyi @ardatan

**Reply by ben-walker** [reply]
Thanks, I'm new to CodeSandbox but I think this link should show the issue I'm having: https://codesandbox.io/s/bold-cdn-t7nf4?file=/codegen.yml

I've set `schema` in `codegen.yml` to the actual GitHub repo I'm trying to pull the schema from, and I get the same "Unable to download schema" error.

**Reply by dotansimha** [reply]
As far as I can see in the implementation of the GitHub loader here: https://github.com/ardatan/graphql-tools/blob/master/packages/loaders/github/src/index.ts#L71-L84 , it's using the GraphQL API of GitHub, to fetch the file data. This API requires a token, so you must provide a access-token for GitHub in order to fetch data, even if the repo is public.

In you specific use-case, you can just do:

```yml
schema: https://raw.githubusercontent.com/pokernook/graph/main/schema.graphql
```

Since the url-loader also knows how to deal with remote SDL files ;) 

@ardatan Maybe we can add support for pulling the file content using the `raw` api of github? so it will allow importing it directly from github without authentication at all.

**Reply by ben-walker** [reply]
Ahh I see, makes sense, thanks! Using raw.githubusercontent seems like the way to go for now, but is there a way I could provide a GitHub token in codegen.yaml if I ever need to go that route?

**Reply by dotansimha** [reply]
Sure. You can do:

```yml
schema:
  - github:pokernook/graph#main:schema.graphql:
      token: XYZ
```
